### PR TITLE
[ch] various ad hoc queries, mostly TD related

### PR DESCRIPTION
**Pull request by clee2000**
* Enables caching for query_clickhouse
* Migrate som adhoc queries to clickhouse, mostly TD related queries
* Some random fixes for some things

### CODE REVIEW

File: tools/torchci/td/get_merge_base_info.py [hunk 1]
```diff
@@ -17,36 +17,24 @@
...
 FAILED_TEST_SHAS_QUERY = """
 SELECT
-    DISTINCT j.head_sha,
+    DISTINCT j.head_sha as head_sha
 FROM
-    commons.failed_tests_run t
-    join workflow_job j on t.job_id = j.id
-    left outer join commons.merge_bases mb on j.head_sha = mb.sha
+    default.failed_test_runs t
+    join default.workflow_job j on t.job_id = j.id
```
Comment (huydhn) [inline tools/torchci/td/get_merge_base_info.py:23]: `final`?

File: tools/torchci/td/get_merge_base_info.py [hunk 2]
```diff
@@ -17,36 +17,24 @@
...
+    left anti join default.merge_bases mb on j.head_sha = mb.sha
```
Comment (huydhn) [inline tools/torchci/td/get_merge_base_info.py:24]: Any insights why `anti join` is a better choice here?

File: tools/torchci/td/utils.py [hunk 3]
```diff
@@ -161,15 +161,15 @@ def get_filtered_failed_tests() -> List[Dict[str, Any]]:
         t.name,
         t.classname,
         t.file,
-        j.head_sha,
+        j.head_sha
     FROM
-        commons.failed_tests_run t
-        join workflow_job j on t.job_id = j.id
+        default .failed_test_runs t
```
Comment (huydhn) [inline tools/torchci/td/utils.py:166]: It looks like a typo in the table name here (has a redundant space)

File: tools/torchci/td/get_merge_base_info.py (hunk 2 above)
Comment (clee2000) [reply tools/torchci/td/get_merge_base_info.py:24]: anti join is an outer join that filters out results where the right side has no match.  I don't know if there are any performance benefits, but it does seem cleaner in terms of query syntax

File: tools/torchci/td/utils.py (hunk 3 above)
Comment (clee2000) [reply tools/torchci/td/utils.py:166]: This is just how they format things, the query still works

File: tools/torchci/td/get_merge_base_info.py (hunk 2 above)
Comment (ZainRizvi) [reply tools/torchci/td/get_merge_base_info.py:24]: That sounds like the same as a "RIGHT JOIN" in normal SQL terminology?

Comment (clee2000) [reply tools/torchci/td/get_merge_base_info.py:24]: Sorry I mean that it only includes results where the right side has no match

File: tools/torchci/td/utils.py (hunk 3 above)
Comment (huydhn) [reply tools/torchci/td/utils.py:166]: TIL, it's a weird SQL syntax

**Pull request merged**